### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by avisil**
Adding detailed comments to the MRC readme.
 - Dataset names with their links
 - small detail about confidence calibration
 - placeholders with todos for score and time logs.

Signed-off-by: Avi Sil <43005718+avisil@users.noreply.github.com>

### CODE REVIEW

File: examples/mrc/README.md [hunk 1]
```diff
@@ -20,6 +23,13 @@ python examples/mrc/run_mrc.py --model_name_or_path xlm-roberta-large \
        --warmup_ratio 0.1 --weight_decay 0.1 --save_steps 50000 \
        --overwrite_output_dir --num_train_epochs 1 --evaluation_strategy no
 ```
+This will detect a GPU if present as well as multiple CPU cores for accelerating preprocessing.
+Some hyperparameters (e.g. fp16, batch size, gradient accumulation steps) may need to be changed
+depending on your hardware configuration.
+
+This yields the following results \TODO:
+```add our results here in the HF format as shown in https://github.com/huggingface/transformers/tree/main/examples/pytorch/question-answering
```
Comment (bhavani105) [inline examples/mrc/README.md:31]: *** / Aggregate Scores ****
{"avg_passage_f1": 0.7214616561954854, "avg_passage_recall": 0.7060810023084223, "avg_passage_precision": 0.7403183125930369, "avg_minimal_f1": 0.6744985613562953, "avg_minimal_recall": 0.6279638519632065, "avg_minimal_precision": 0.733070067175786}

***** eval metrics *****
  epoch                      =    1.0
  eval_avg_minimal_f1        = 0.6745
  eval_avg_minimal_precision = 0.7331
  eval_avg_minimal_recall    =  0.628
  eval_avg_passage_f1        = 0.7215
  eval_avg_passage_precision = 0.7403
  eval_avg_passage_recall    = 0.7061
  eval_samples               = 238831

**Review by bhavani105 — COMMENTED** [review]
TyDi eval metrics are available here:
https://github.ibm.com/ai-foundation/PrimeQA/issues/230

File: examples/mrc/README.md (hunk 1 above)
Comment (avisil) [reply examples/mrc/README.md]: Added it. thx

**Review by vishwajeetkumar93 — APPROVED** [review]

**Comment by yulongl-ibm** [comment]
@jsmcibm 
In https://github.com/primeqa/primeqa/tree/updated-documentation-readme/examples/boolqa there is:
Machine Reading Comprehension: given a question and answer, find a representative span that may contain a short answer. This is analyzed in detail in the tydiqa.ipynb

Maybe add a link to the tydiqa.ipynb?

**Comment by yulongl-ibm** [comment]
@mihaela-bornea 
There is a TODO in https://github.com/primeqa/primeqa/blob/updated-documentation-readme/examples/mrc/README.md :
if you want to do [confidence calibration (https://arxiv.org/abs/2101.07942) (\TODO: add a figure) estimate of your fine-tuned model use the following:

Delete it or finish it?

**Review by yulongl-ibm — APPROVED** [review]

**Review by yulongl-ibm — CHANGES_REQUESTED** [review]
@franzmpub 
In https://github.com/primeqa/primeqa/blob/2d7640037ddedadea1f3301a99608e4a3e2ec222/notebooks/ir/dense/dense_ir_student_teacher.ipynb

I think this section you meant distillation with triples instead of parallel data?

:Next, we run Knowledge Distillation using parallel data containing:
a) non-English queries and English passages for the student model
b) English queries and English passages for the student model.

File: examples/ir/README.md [hunk 2]
```diff
@@ -1,3 +1,106 @@
+## Dense retrieval
+
+We support ColBERT-based Information Retrieval engine, as described in [README](../../primeqa/ir/dense/colbert_top/README.md).
+
+This README shows how to run the basic model training, data indexing, retrieval, and scoring steps, using the `run_ir.py` script.  The steps involved in training a model using the DR.DECR (Dense Retrieval with Distillation-Enhanced Cross-Lingual Representation) Student/Teacher pipeline, as desribed in [Learning Cross-Lingual IR from an English Retriever](https://arxiv.org/abs/2112.08185), are outlined in the [Jupyter notebook](../../notebooks/ir/dense/dense_ir_student_teacher.ipynb).
+
+
+The following steps require to have PrimeQA [installed](../../README.md#Installation).
```
Comment (jsmcibm) [inline examples/ir/README.md:8]:
```suggestion
The following steps require PrimeQA to be [installed](../../README.md#Installation).
```

File: examples/ir/README.md [hunk 3]
```diff
@@ -1,3 +1,106 @@
...
+Sample data files are [here](../../tests/resources/ir_dense), their formats are shown in the Jupyter notebooks [here](../../notebooks/ir/dense/dense_ir.ipynb) and [here](../../notebooks/ir/dense/dense_ir_student_teacher.ipynb).
+
+### Model Training
+
+Here is an example of an training run for a Question Anwering model, using a training data .tsv file containing training examples in the form of \<query>, <positive_passage>, <negative_passage> triples.
```
Comment (jsmcibm) [inline examples/ir/README.md:13]:
```suggestion
Here is an example of a training run for a Question Anwering model, using a training data .tsv file containing training examples in the form of \<query>, <positive_passage>, <negative_passage> triples.
```

**Review by jsmcibm — APPROVED** [review]
(1) in `https://github.com/primeqa/primeqa/blob/main/examples/mrc/README.md`, language codes are written uppercase - I think language codes are usually lowercase.
(2) the MRC notebook is not rendering in safari - I've already brought this to Avi's attention
(3) there should be a discussion about the number of significant figures when reporting results - some of our results are reported more accurately than the V100 vs A100 difference

File: examples/mrc/README.md [hunk 4]
```diff
@@ -27,42 +49,45 @@ python examples/mrc/run_mrc.py --model_name_or_path xlm-roberta-large \
...
-For eval-only TyDiQA with support for boolean questions (for [details](../boolqa/README.md)):
-```shell
-python examples/mrc/run_mrc.py --model_name_or_path ibm/tydiqa-primary-task-xlm-roberta-large \
-       --output_dir ${OUTPUT_DIR} --fp16 \
-       --per_device_eval_batch_size 128 --overwrite_output_dir \
-       --do_boolean --boolean_config  examples/boolqa/tydi_boolqa_config.json
+       --per_device_eval_batch_size 128 --overwrite_output_dir --overwrite_cache
 ```
 
+- if you want to do [confidence calibration](https://arxiv.org/abs/2101.07942) (\TODO: add a figure) estimate of your fine-tuned model use the following:
```
[the remark this answers is not in the document]
Comment (avisil) [reply examples/mrc/README.md]: fixed.
